yt_dlp/extractor/pbs.py

```diff
@@ -622,8 +622,9 @@ def extract_redirect_urls(info):
                 continue
 
             if determine_ext(format_url) == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
-                    format_url, display_id, 'mp4', m3u8_id='hls', fatal=False))
+                hls_formats, _ = self._extract_m3u8_formats_and_subtitles(
```

Comment:
`_extract_m3u8_formats` was giving warnings regarding there being hls subtitles. but when extracted it just gives the .vtt filename. I'm not familiar with this area so used `_extract_m3u8_formats_and_subtitles` and left the extracted subtitles alone for now.

Reply:
Now that I think about it this is expected behaviour, yt-dlp handles it correctly. Just that the test I copied over doesn't work with it. 

Adding it back would mean duplicate vtt subtitle, but don't think that should be an issue?

Reply:
How about keeping WebVTT extraction here and removing the string-substitution of the TTML URL later? Especially if those methods give the exact same URL currently.

Extracting subtitles from a standardised manifest should be more robust than blindly assuming it will be available at a link obtained by fragile string-substitution on some other known URI, as the latter may change at the website owner’s whim. Plus extracting from a manifest is going to keep working transparently if they switch to another subtitle format (e.g. plain WebVTT to segmented WebVTT)